src/page-merger.js

```diff
@@ -0,0 +1,62 @@
+class PageMerger {
+  constructor(paginator) {
+    this._count = null;
+    this._paginator = paginator;
+  }
+
+  merge(startPage, endPage) {
+    return this._fetchRequests(startPage, endPage)
+      .then(this._makeCollection.bind(this))
+      .then(this._mergeCollection.bind(this))
+      .then(this._makeResponse.bind(this));
+  }
+
+  _fetchRequests(startPage, endPage) {
+    const paginator = this._paginator;
+    const requests = [];
+
+    for (let page = startPage; page <= endPage; page++) {
+      requests.push(paginator.fetchPage(page));
+    }
+
+    return Promise.all(requests);
+  }
+
+  _makeCollection(responses) {
+    const collection = [];
+
+    for (let i = 0; i < responses.length; i++) {
```

Comment:
This is a good opportunity for `reduce`, either native or lodash's implementation.